rust/nasl-builtin-regex/src/lib.rs

```diff
@@ -0,0 +1,256 @@
+/// - icase   case insensitive flag
+/// - rnul    replace the null char in the string. Default TRUE.
+/// - multiline Is FALSE by default (string is truncated at the first
+///                “end of line”), and can be set to TRUE for multiline search.
+/// Return true if matches, false otherwise
+#[nasl_function(named(string, pattern, icase, rnul, multiline))]
+fn ereg(
+    string: NaslValue,
+    pattern: NaslValue,
+    icase: Option<bool>,
+    rnul: Option<bool>,
+    multiline: Option<bool>,
+) -> Result<NaslValue, FunctionErrorKind> {
+    let icase = icase.unwrap_or_default();
+    let rnul = rnul.unwrap_or(true);
+    let multiline = multiline.unwrap_or_default();
+
+    let str_bind = string.to_string();
+    let mut str_aux = str_bind.as_str();
+
+    if !rnul {
+        let s = match str_aux.chars().position(|c| c == '\0') {
+            Some(i) => i,
+            None => str_aux.len(),
+        };
+        (str_aux, _) = str_aux.split_at(s);
+    }
+
+    if !multiline {
+        let s = match str_aux.chars().position(|c| c == '\n') {
+            Some(i) => i,
+            None => str_aux.len(),
+        };
+        (str_aux, _) = str_aux.split_at(s);
+    }
+
+    let re = match RegexBuilder::new(pattern.to_string().as_str())
+        .case_insensitive(icase)
+        .multi_line(multiline)
+        .build()
+    {
+        Ok(re) => re,
+        Err(e) => {
+            return Err(FunctionErrorKind::Dirty(format!(
+                " Error building regular expression pattern: {}",
+                e
+            )));
+        }
+    };
+
+    Ok(NaslValue::Boolean(re.is_match(str_aux)))
+}
+
+/// Search for a pattern in a string and replace it.
+/// - string  String to search the pattern in
+/// - pattern pattern to search in the string for
+/// - replace string to replace the pattern with
+/// - icase   case insensitive flag
+/// - rnul    replace the null char in the string. Default TRUE.
+///
+/// Return the new string with the pattern replaced with replace.
+#[nasl_function(named(string, pattern, replace, icase, rnul))]
+fn ereg_replace(
+    string: NaslValue,
+    pattern: NaslValue,
+    replace: NaslValue,
+    icase: Option<bool>,
+    rnul: Option<bool>,
+) -> Result<NaslValue, FunctionErrorKind> {
+    let icase = icase.unwrap_or_default();
+    let rnul = rnul.unwrap_or(true);
+
+    let str_bind = string.to_string();
+    let mut str_aux = str_bind.as_str();
+
+    if !rnul {
+        let s = match str_aux.chars().position(|c| c == '\0') {
+            Some(i) => i,
+            None => str_aux.len(),
+        };
+        (str_aux, _) = str_aux.split_at(s);
+    }
+
+    let re = match RegexBuilder::new(pattern.to_string().as_str())
+        .case_insensitive(icase)
+        .build()
+    {
+        Ok(re) => re,
+        Err(e) => {
+            return Err(FunctionErrorKind::Dirty(format!(
+                " Error building regular expression pattern: {}",
+                e
+            )));
+        }
+    };
```

Comment:
Similar logic also appears many times throughout the code. Please extract it into a function so that it becomes just

```rust
fn make_regex(pattern: &str, icase: bool, multiline: bool) -> Result<Regex, FunctionErrorKind> {
    ...
}
```

which can then simply be called as
```
make_regex(pattern, icase, multline)?
```